sympy/parsing/sympy_parser.py

```diff
@@ -1061,8 +1061,7 @@ def visit_BinOp(self, node):
             sympy_class = self.operators[node.op.__class__]
             right = self.visit(node.right)
             left = self.visit(node.left)
-            if isinstance(node.left, ast.UnaryOp) and (isinstance(node.right, ast.UnaryOp) == 0) and sympy_class in ('Mul',):
-                left, right = right, left
```

Comment:
I'm not sure I understand why this happens. Tracing through the blame this seems to come from https://github.com/sympy/sympy/pull/10774

@smichr do you have any idea why these are swapped?

Reply:
Must have been a work around for something that has since been fixed. I no longer recall any specifics other than what was recorded in the PR that did this.

Reply:
Actually I think maybe the swap here is correct and shouldn't be removed.

The linked issue is to do with the printer not the parser: https://github.com/sympy/sympy/issues/21119#issuecomment-809534656

Reply:
But with this PR the following regresses:

in master
```python
>>> sympify('-10/5', evaluate = False)
1/5*(-10)
>>> _.args
(1/5, -10)
```
 now gives the incorrect result,
```
>>> sympify('-10/5', evaluate = False).args
(1/(-10), 5)
```

Reply:
And I am not sure what the OP comment means in light of the following correct parse in master:
```
>>> S('9.2*(4.1+(-4)*3)',evaluate=0)
9.2*(4.1 + (-4)*3)
```

Reply:
> linked issue is to do with the printer not the parser

I think there is a parser issue:
```
>>> S('3*4',evaluate=0).args
(3, 4)
>>> S('3*-4',evaluate=0).args
(3, -4)
>>> S('-3*4',evaluate=0).args
<reversal code traversed>
(4, -3)
>>> S('-3*-4',evaluate=0).args
(-3, -4)

>>> S('1/2',evaluate=0).args
(1, 1/2)
>>> S('1/-2',evaluate=0).args
(1, 1/(-2))
>>> S('-1/2',evaluate=0).args
<reversal code traversed>
(1/2, -1)
>>> S('-1/-2',evaluate=0).args
(1/(-2), -1)
```

Reply:
@smichr Right, missed the `Div` case in here, will update.
I think we can move that swap in the `ast.Div` if statement itself, since doesn't seem to be needed for `ast.Mul` operator  like this.
```
--- a/sympy/parsing/sympy_parser.py
+++ b/sympy/parsing/sympy_parser.py
@@ -1062,7 +1062,12 @@ def visit_BinOp(self, node):
             sympy_class = self.operators[node.op.__class__]
             right = self.visit(node.right)
             left = self.visit(node.left)
-            if isinstance(node.left, ast.UnaryOp) and (isinstance(node.right, ast.UnaryOp) == 0) and sympy_class in ('Mul',):
-                left, right = right, left
+            rev = False
             if isinstance(node.op, ast.Sub):
                 right = ast.Call(
                     func=ast.Name(id='Mul', ctx=ast.Load()),
@@ -1073,8 +1078,8 @@ def visit_BinOp(self, node):
                 )
             if isinstance(node.op, ast.Div):
                 if isinstance(node.left, ast.UnaryOp):
-                    if isinstance(node.right,ast.UnaryOp):
-                        left, right = right, left
+                    left, right = right, left
+                    rev = True
                     left = ast.Call(
                     func=ast.Name(id='Pow', ctx=ast.Load()),
                     args=[left, ast.UnaryOp(op=ast.USub(), operand=ast.Num(1))],
@@ -1091,6 +1096,8 @@ def visit_BinOp(self, node):
                     kwargs=None
                 )

+            if rev:  # undo reversal
+                left, right = right, left
```